corehq/apps/sso/forms.py

```diff
@@ -631,6 +717,37 @@ def clean_entity_id(self):
         _ensure_entity_id_matches_expected_provider(entity_id, self.idp)
         return entity_id
 
+    def clean_api_secret(self):
+        api_secret = self.cleaned_data['api_secret']
+        is_enabled = self.cleaned_data['enable_user_deactivation']
+        if is_enabled and (not api_secret and not self.idp.api_secret):
```

Comment:
```suggestion
        if is_enabled and not (api_secret or self.idp.api_secret):
```
this feels more readable?